hikari/impl/cache.py

```diff
@@ -820,8 +820,6 @@ def _garbage_collect_member(
             guild_record.members = None
             self._remove_guild_record_if_empty(member.object.guild_id, guild_record)
 
-        return member
```

Comment:
You should just change this to return the object if it wasn't originally marked as has_been_deleted

Reply:
The reason I changed this to not return anything is because there is no need, as the information we return in the same as the one we take in. The cases we use this for atm don't even need what we return

Reply:
Clear and delete methods should only be returning objects which were actually marked as deleted not objects which are being kept alive by references, the way you've changed the semantics still leads to erroneous behaviour and simply changing the behaviour of the garbage collect method will instruct both the delete and clear methods on what they should be saying was deleted and avoids a ton of other code changes